Send invoice and testing

NickCoding22 opened the pull request.

skberkeley committed:
Tests for QuickBooksModule constructor and getAllCustomers

NickCoding22 committed:
Completed sendInvoice and implemented associated testing methods.

skberkeley commented on app/src/test/java/invoice_automation/module/QuickBooksModuleTest.java:
```diff
@@ -75,6 +80,11 @@ private void setupMethodTests() throws Exception {
 
+    private void setupInvoiceTests() {
+        invoice = new Invoice();
+        invoice.setBillEmail(billEmail);
```
Let's move this line to each unit tests which actually requires the billing email to be set, since future invoice tests don't necessarily need `billEmail` to be set

skberkeley commented on app/src/main/java/invoice_automation/module/QuickBooksModule.java:
```diff
@@ -103,7 +104,12 @@ public Map<InvoiceType, Invoice> getOrCreateInvoicesFromRegistration(@NonNull Re
+        } catch (FMSException e) {
+            throw new QuickBooksException("Exception sending invoice", e);
+        }
+        invoice.setEmailStatus(EmailStatusEnum.EMAIL_SENT);
```
I'm curious, is this line needed?

NickCoding22 replied:
Yeah, you need to change the status manually, sendEmail doesn't alter it

skberkeley commented on app/src/test/java/invoice_automation/module/QuickBooksModuleTest.java:
```diff
@@ -97,4 +107,28 @@ public void testGetAllCustomers_dataServiceThrows() throws Exception {
+        quickBooksModule.sendInvoice(invoice);
+
+        // Verify
+        assertEquals(EmailStatusEnum.EMAIL_SENT, invoice.getEmailStatus());
```
We should also check that the DataService's sendEmail method got called, since all this checks right now is whether the email status was changed to sent.

skberkeley reviewed (changes requested):
Can you rebase your changes onto main?

Can you also manually test by:

1. Setting up an Intuit dev account and creating a sandbox company
2. Create a new customer in the sandbox company with an email you have access to
3. Create an invoice to that customer
4. Modify App's main method to access this invoice (check Intuit's documentation on how to do queries), and send it
5. Run main (don't forgot to update the access tokens by getting them from Intuit's authentication playground) and verify that the invoice was emailed properly

skberkeley committed:
Verified that dataService.sendEmail is called

NickCoding22 committed:
sendInvoice test now checks that the email has been sent and a manual test has been added to App.

skberkeley committed:
Merge remote-tracking branch 'origin/send_invoice_and_testing' into send_invoice_and_testing

skberkeley committed:
Rewrote getTestInvoice to getInvoiceWithMatchingMemo in QuickBooksModule
Added tests for getInvoiceWithMatchingMemo in QuickBooksModuleTest
Manually tested

skberkeley commented:
Cherrypicked changes onto a different branch
